src/xdg.c

```diff
@@ -175,17 +175,33 @@ handle_request_maximize(struct wl_listener *listener, void *data)
 		/*store_natural_geometry*/ true);
 }
 
+static void
+set_fullscreen_from_request(struct view *view,
+		struct wlr_xdg_toplevel_requested *requested)
+{
+	if (!view->fullscreen && requested->fullscreen
+			&& requested->fullscreen_output) {
+		struct output *output = output_from_wlr_output(view->server,
+			requested->fullscreen_output);
+		if (output_is_usable(output)) {
+			view->output = output;
```

Comment:
This directly modifies the output for the view, what do you think about something like `view_set_output(view, output)` instead?

Reply:
`view_set_output(view, output)` might be nicer.

Inside `view_set_output`, we could do `assert(!view->fullscreen)` (or maybe just return) since modifying `view->output` for a fullscreen view isn't allowed.